eng/Versions.props

```diff
@@ -19,7 +19,8 @@
     <UsingToolIbcOptimization>true</UsingToolIbcOptimization>
     <UsingToolXliff>false</UsingToolXliff>
     <!-- Downgrade compiler version to fix nullability issues: https://github.com/dotnet/runtime/issues/34417. -->
-    <MicrosoftNetCompilersToolsetVersion>3.6.0-2.20166.2</MicrosoftNetCompilersToolsetVersion>
+    <!-- <MicrosoftNetCompilersToolsetVersion>3.6.0-2.20166.2</MicrosoftNetCompilersToolsetVersion> -->
```

Comment:
Reminder: I plan on reverting the changes to this file if possible. It's only here to demonstrate which toolsets we're using.

Reply:
Any reason we can't keep the newest version of the compiler?  I'm unclear why we haven't taken a newer one recently.

Reply:
Our props file explicitly pinned to an older version partially due to breaking changes in how nullability annotations are processed. It also turns out there's a bug in the compiler toolset version currently carried by Arcade (see https://github.com/dotnet/roslyn/issues/44000) that causes build failures. That bug was addressed in the latest nightly compiler toolchain, which is why for this PR I've leapfrogged the toolset version that comes with Arcade and instead pinned the latest fixed version.

Reply:
> I've leapfrogged the toolset version that comes with Arcade and instead pinned the latest fixed version.

Ok, thanks.  I didn't understand the "I plan on reverting the changes to this file" part.  I thought you were saying you were making the changes to the annotations but wouldn't actually check in the switch to the newer compiler.  Since we were already pinning a version, I suggest you keep your change to pin the latest version that works.  If that's what you were already planning on doing, great :)

Reply:
I \[not-so\] secretly just wanted the nint/nuint support that the latest toolchain brings. ;)